beluga/include/beluga/estimation.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022-2023 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ESTIMATION_HPP
+#define BELUGA_ESTIMATION_HPP
+
+/**
+ * \file
+ * \brief Implementation of algorithms that allow calculating the true state estimator of a particle set.
+ */
+
+/**
+ * \page StateEstimatorPage Beluga named requirements: StateEstimator
```

Comment:
I would prefer the different algorithms to be separate function overloads rather than classes:

```cpp
beluga::estimate(states);  // simple
beluga::weighted_estimate(states, weights);
beluga::weighted_estimate(particles);
beluga::cluster_based_estimate(states, weights);
beluga::cluster_based_estimate(particles);
```